ArduPlane/landing.cpp

```diff
@@ -25,6 +25,26 @@ bool Plane::verify_land()
         return true;
     }
 
+    // when aborting a landing, mimic the verify_takeoff with steering hold. Once
+    // the altitude has been reached, restart the landing sequence
+    if (flight_stage == AP_SpdHgtControl::FLIGHT_LAND_ABORT) {
+
+        throttle_suppressed = false;
+        auto_state.land_complete = false;
+        nav_controller->update_heading_hold(steer_state.hold_course_cd);
+
+        // see if we have reached abort altitude
+        if (adjusted_relative_altitude_cm() > auto_state.takeoff_altitude_rel_cm) {
+            steer_state.hold_course_cd = -1;
+            next_WP_loc = current_loc;
+            mission.stop();
+            restart_landing_sequence();
```

Comment:
if the restart of the landing sequence fails won't we end up flying off forever?
I think if it fails we probably need to RTL. Either that or we work out what we are going to do before we start the abort, and don't abort the landing if we don't know for sure we can restart the landing sequence